bench/rttanalysis: TestBenchmarkExpectationShard2 failed [s390x: SIGSEGV in runtime.(*mheap).freeManual]

bench/rttanalysis.TestBenchmarkExpectationShard2 [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20512433?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20512433?buildTab=artifacts#/) on release-25.3 @ [f3b11227b9b240d776e19fcd1bf49592de5605e5](https://github.com/cockroachdb/cockroach/commits/f3b11227b9b240d776e19fcd1bf49592de5605e5):


```
github.com/cockroachdb/cockroach/pkg/server.(*serverController).sqlMux(0xc0081189a0, {0x86b39e80, 0xc056ff32c0}, {0x86b79110, 0xc02f353888}, {0x0, 0x4, 0x0, 0xc059925bf0, {{{{...}}, ...}, ...}})
	pkg/server/server_controller_sql.go:90 +0x310 fp=0xc00ee999f8 sp=0xc00ee997c0 pc=0x84e3ac20
github.com/cockroachdb/cockroach/pkg/server.(*serverController).sqlMux-fm({0x86b39e80, 0xc056ff32c0}, {0x86b79110, 0xc02f353888}, {0x0, 0x4, 0x0, 0xc059925bf0, {{{{...}}, 0x0, ...}, ...}})
	<autogenerated>:1 +0x90 fp=0xc00ee99af8 sp=0xc00ee999f8 pc=0x84ed6a00
github.com/cockroachdb/cockroach/pkg/server.startServeSQL.func1.1({0x86b39e80, 0xc0902d23c0}, {0x86b781e0, 0xc061e6ecb8})
	pkg/server/server_sql.go:1994 +0x272 fp=0xc00ee99ef0 sp=0xc00ee99af8 pc=0x84e4c862
github.com/cockroachdb/cockroach/pkg/util/netutil.(*TCPServer).ServeWith.func1({0x86b39e80, 0xc0902d23c0})
	pkg/util/netutil/net.go:197 +0x142 fp=0xc00ee99fc0 sp=0xc00ee99ef0 pc=0x8212dc12
github.com/cockroachdb/cockroach/pkg/util/netutil.(*TCPServer).ServeWith.gowrap1()
	pkg/util/netutil/net.go:198 +0x70 fp=0xc00ee99fd8 sp=0xc00ee99fc0 pc=0x8212dab0
runtime.goexit({})
	src/runtime/asm_s390x.s:797 +0x2 fp=0xc00ee99fd8 sp=0xc00ee99fd8 pc=0x800b1992
created by github.com/cockroachdb/cockroach/pkg/util/netutil.(*TCPServer).ServeWith in goroutine 473812
	pkg/util/netutil/net.go:190 +0x398

goroutine 464834 gp=0xc062bc3dc0 m=nil [select, 1 minutes]:
runtime.gopark(0x8625f140, 0x0, 0x9, 0x3, 0x1)
	GOROOT/src/runtime/proc.go:424 +0x136 fp=0xc00e27cd50 sp=0xc00e27cd38 pc=0x800a7666
runtime.selectgo(0xc00e27cf00, 0xc0952ffee8, 0x0, 0x0, 0x2, 0x1)
	GOROOT/src/runtime/select.go:335 +0x8ac fp=0xc00e27cea0 sp=0xc00e27cd50 pc=0x800792fc
github.com/cockroachdb/cockroach/pkg/sql.(*runnerCoordinator).init.func1({0x86b39e80, 0xc00b69b900})
	pkg/sql/distsql_running.go:164 +0xba fp=0xc00e27cf50 sp=0xc00e27cea0 pc=0x843a778a
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0x86b39e80, 0xc00b69b900})
	pkg/util/stop/stopper.go:438 +0xac fp=0xc00e27cfc0 sp=0xc00e27cf50 pc=0x8128caac
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.gowrap1()
	pkg/util/stop/stopper.go:439 +0x70 fp=0xc00e27cfd8 sp=0xc00e27cfc0 pc=0x8128c9e0
runtime.goexit({})
	src/runtime/asm_s390x.s:797 +0x2 fp=0xc00e27cfd8 sp=0xc00e27cfd8 pc=0x800b1992
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 464224
	pkg/util/stop/stopper.go:436 +0x16c

r0   0x0	r1   0x0
r2   0x0	r3   0x3ff4bafebf8
r4   0x108	r5   0xc002a801c0
r6   0xc000003180	r7   0xc072587fff
r8   0x0	r9   0x18
r10  0x8a125580	r11  0x3ff4c2ff900
r12  0xc0000c56d0	r13  0xc002a801c0
r14  0x80046ad8	r15  0x3ff4c2fe3f8
pc   0x8004aaf0	link 0x80046ad8
--- FAIL: TestBenchmarkExpectationShard2 (263.82s)
=== RUN   TestBenchmarkExpectationShard2/ORMQueries
=== RUN   TestBenchmarkExpectationShard2/ORMQueries/column_descriptions_json_agg
=== RUN   TestBenchmarkExpectationShard2/UseManyRoles/use_50_roles
=== RUN   TestBenchmarkExpectationShard2/UseManyRoles
=== RUN   TestBenchmarkExpectationShard2/ORMQueries/liquibase_migrations
=== RUN   TestBenchmarkExpectationShard2/ORMQueries/liquibase_migrations_on_multiple_dbs
=== RUN   TestBenchmarkExpectationShard2/ORMQueries/prisma_column_descriptions_updated
=== RUN   TestBenchmarkExpectationShard2/ORMQueries/prisma_types_16
=== RUN   TestBenchmarkExpectationShard2/ORMQueries/prisma_types_4
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBenchmarkExpectationShard2.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>



Jira issue: CRDB-54799

--- a/pkg/bench/rttanalysis/BUILD.bazel
+++ b/pkg/bench/rttanalysis/BUILD.bazel
@@ -13,8 +13,6 @@ go_library(
     visibility = ["//visibility:public"],
     deps = [
         "//pkg/base",
-        "//pkg/jobs",
-        "//pkg/jobs/jobspb",
         "//pkg/kv/kvclient/kvcoord",
         "//pkg/sql",
         "//pkg/sql/parser",
@@ -58,9 +56,9 @@ go_test(
     data = glob(["testdata/**"]),
     embed = [":rttanalysis"],
     exec_properties = {"test.Pool": "large"},
-    shard_count = 4,
     deps = [
         "//pkg/base",
+        "//pkg/jobs",
         "//pkg/jobs/jobspb",
         "//pkg/security/securityassets",
         "//pkg/security/securitytest",
@@ -71,7 +69,6 @@ go_test(
         "//pkg/testutils/serverutils",
         "//pkg/testutils/skip",
         "//pkg/testutils/testcluster",
-        "//pkg/util/envutil",
         "//pkg/util/protoutil",
         "//pkg/util/randutil",
     ],

--- a/pkg/bench/rttanalysis/registry.go
+++ b/pkg/bench/rttanalysis/registry.go
@@ -9,8 +9,6 @@ import (
 	"strings"
 	"testing"
 
-	"github.com/cockroachdb/cockroach/pkg/jobs"
-	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
 	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/errors"
 	"github.com/stretchr/testify/require"
@@ -53,66 +51,15 @@ func (r *Registry) Run(b *testing.B) {
 // benchmarks can be filtered by passing the usual test filters underneath
 // this test's name.
 //
-// It takes a long time and thus is skipped under duress and short.
+// It takes a long time and thus is skipped under stress, race
+// and short.
 func (r *Registry) RunExpectations(t *testing.T) {
-	r.RunExpectationsSharded(t, 1, 1)
-}
-
-// RunExpectationsSharded runs all the benchmarks for one iteration
-// and validates that the number of RPCs meets the expectation. If run
-// with the --rewrite flag, it will rewrite the run benchmarks. The
-// benchmarks can be filtered by passing the usual test filters underneath
-// this test's name.
-//
-// It takes a long time and thus is skipped under duress and short.
-//
-// When shard and totalShards are provided (> 1), only a subset of benchmarks
-// assigned to the specific shard will be run, enabling parallel execution.
-// Test groups are distributed across shards using round-robin assignment.
-func (r *Registry) RunExpectationsSharded(t *testing.T, shard, totalShards int) {
-	defer jobs.TestingSetIDsToIgnore(map[jobspb.JobID]struct{}{3001: {}, 3002: {}})()
-	skip.UnderDuress(t)
+	skip.UnderStress(t)
+	skip.UnderRace(t)
 	skip.UnderShort(t)
+	skip.UnderDeadlock(t)
 
-	// If totalShards is 1, run all tests; otherwise shard them
-	var registryToUse *Registry
-	if totalShards <= 1 {
-		// Run all test groups
-		registryToUse = r
-	} else {
-		// Create a registry with only the test groups assigned to this shard
-		shardRegistry := &Registry{
-			numNodes: r.numNodes,
-			cc:       r.cc,
-			r:        make(map[string][]RoundTripBenchTestCase),
-		}
-
-		// Distribute test groups across shards using round-robin assignment
-		// First, get all group names and sort them for consistent ordering
-		groupNames := make([]string, 0, len(r.r))
-		for groupName := range r.r {
-			groupNames = append(groupNames, groupName)
-		}
-		// Sort for deterministic assignment across runs
-		for i := 0; i < len(groupNames); i++ {
-			for j := i + 1; j < len(groupNames); j++ {
-				if groupNames[i] > groupNames[j] {
-					groupNames[i], groupNames[j] = groupNames[j], groupNames[i]
-				}
-			}
-		}
-
-		// Assign groups to shards using round-robin
-		for i, groupName := range groupNames {
-			assignedShard := (i % totalShards) + 1
-			if assignedShard == shard {
-				shardRegistry.r[groupName] = r.r[groupName]
-			}
-		}
-		registryToUse = shardRegistry
-	}
-
-	runBenchmarkExpectationTests(t, registryToUse)
+	runBenchmarkExpectationTests(t, r)
 }
 
 // Register registers a set of test cases to a given benchmark name. It is

--- a/pkg/bench/rttanalysis/validate_benchmark_data_test.go
+++ b/pkg/bench/rttanalysis/validate_benchmark_data_test.go
@@ -6,44 +6,13 @@
 package rttanalysis
 
 import (
-	"strconv"
 	"testing"
 
-	"github.com/cockroachdb/cockroach/pkg/util/envutil"
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
 )
 
-// NOTE: If you change the number of shards, you must also update the
-// shard_count in BUILD.bazel to match.
-const shardCount = 4
-
-// Validate that shardCount matches TEST_TOTAL_SHARDS environment variable at init time
-var _ = func() int {
-	totalShardsStr, found := envutil.ExternalEnvString("TEST_TOTAL_SHARDS", 1)
-	if totalShardsStr == "" || !found {
-		return 0
-	}
-	totalShards, err := strconv.Atoi(totalShardsStr)
-	if err != nil {
-		return 0
-	}
-	if totalShards != shardCount {
-		panic("shardCount mismatch: update shard_count in pkg/bench/rttanalysis/BUILD.bazel to match shardCount constant")
-	}
-	return 0
-}()
-
-func TestBenchmarkExpectationShard1(t *testing.T) {
-	reg.RunExpectationsSharded(t, 1, shardCount)
-}
-
-func TestBenchmarkExpectationShard2(t *testing.T) {
-	reg.RunExpectationsSharded(t, 2, shardCount)
-}
-
-func TestBenchmarkExpectationShard3(t *testing.T) {
-	reg.RunExpectationsSharded(t, 3, shardCount)
-}
-
-func TestBenchmarkExpectationShard4(t *testing.T) {
-	reg.RunExpectationsSharded(t, 4, shardCount)
+func TestBenchmarkExpectation(t *testing.T) {
+	defer jobs.TestingSetIDsToIgnore(map[jobspb.JobID]struct{}{3001: {}, 3002: {}})()
+	reg.RunExpectations(t)
 }

